pkg/network/namescheme/networknamescheme.go

```diff
@@ -92,17 +92,19 @@ func CreateOrdinalNetworkNameScheme(vmiNetworks []v1.Network) map[string]string
 	return networkNameSchemeMap
 }
 
+// OrdinalPodInterfaceName returns the ordinal interface name for the given network name.
+// networks contains pod networks and multus networks.
+// Primary network will use "eth0" and the secondary ones will use "net<id>" format, where id is an enumeration
+// from 1 to n.
 func OrdinalPodInterfaceName(name string, networks []v1.Network) string {
```

Comment:
Thanks for the great catch!
Can you please reuse `CreateOrdinalNetworkNameScheme`, it has the exact same logic.

```
func OrdinalPodInterfaceName(name string, networks []v1.Network) string {
	networkNameSchemeMap := CreateOrdinalNetworkNameScheme(networks)
	if ordinalName, exist := networkNameSchemeMap[name]; exist {
		return ordinalName
	}
	return ""
}
```

Reply:
This function was introduced to avoid creating the whole name scheme for all interfaces, and easier to work with at places the whole name-scheme is not necessary.

 Which this change I think it better to use the map directly where needed (using `CreateOrdinalNetworkNameScheme`) and remove this function (`OrdinalPodInterfaceName`) altogether.
 Or change the mentioned function to use the exact same logic, including the interface index (not sure its possible).

It doesnt have to be on this PR though, so its not a must.

Reply:
Hi @ormergi, 

Sorry, I just misunderstood your meaning.
So, as you mentioned, this function should be simple and avoid building the whole name-scheme map, right?

If yes, I thought we could change back to the original. Just walk through the network slice and encounter the correct index.

WDYT?
@AlonaKaplan

Reply:
You have an lgtm and approve on this PR, I suggest you leave it as is.

Reply:
Thanks @EdDev 
If needed, I could do more with that.

just let me know : )